documentation/library_guide/parallel_api/additional_algorithms.rst

```diff
@@ -76,10 +76,11 @@ header.  All algorithms are implemented in the ``oneapi::dpl`` namespace.
     search sequence: [0, 2, 4, 7, 6]
     result sequence: [1, 4, 8, 10, 10]
 
-* ``sort_by_key``: performs a key-value sort. The algorithm sorts a sequence of keys according to
-  a comparison operator. If no comparator is provided, then the elements are compared with ``operator<``.
-  A sequence of values is simultaneously permuted according to the sorted order of keys. For correct behavior,
-  both sequences must be of the same size.
+* ``sort_by_key``: performs a key-value sort.
+  The algorithm sorts a sequence of keys based on a comparison function object.
+  If it is not provided, the elements are compared with ``operator<``.
+  A sequence of values is simultaneously permuted according to the sorted order of keys.
+  For correct behavior, a sequence of values must have at least as many elements as the sequence of keys.

```

Comment:
I suggest following sentence:
`A sequence of values must have at least as many elements as the sequence of keys, otherwise the behavior is undefined.`

`the behavior is undefined` - "standard" wording on cppreference, f.e., more over we also use one in other parts of the documentation.

Reply:
May be some `assert` required for this check or `exception` ?

Reply:
We cannot check it, because for "values" we have only "begin"  iterator.

Reply:
I've applied the suggestion, with slight changes proposed by Alexey below.